### PR TITLE
chore: tweak prettier rules

### PR DESCRIPTION
**Pull request by felicia-haggqvist**
**Note:** 
I have tested that these changes work in @warp-ds/react, @warp-ds/vue and @warp-ds/elements.

Part of fixing: [WARP-446](https://nmp-jira.atlassian.net/browse/WARP-446)

- Adjust `printWidth` to have `140` instead of `999`
- Add the `eslint-plugin-import` to the index.js-file
- Add `eslint-plugin-import` as a peerDependency

**Note:**
Previously, the idea was to add [@serverless-guru/prettier-plugin-import-order](https://github.com/serverless-guru/prettier-import-order), where it formatted correctly when linking this branch with the @warp-ds/react repo. However, when testing to link this branch with the @warp-ds/vue repo, an error was thrown due to a mismatch between vue-version and the version of vue/compiler-sfc. The only way to fix this error was to change the vue-version from "3.4.19" to "3.3.13", which did not seem like a valid solution, since it would hinder us from bumping vue-version in the future. 

Therefore, I suggest that we instead use the `eslint-plugin-import` that still formats the imports but doesn't throw this error in the @warp-ds/vue-repo. I have also tested to link this branch with @warp-ds/react, @warp-ds/vue and @warp-ds/elements and I've tested to run `pnpm lint:check` and `pnpm lint`.

### CODE REVIEW

**Review by imprashast — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -53,8 +53,20 @@ module.exports = {
       'warn',
       {
         singleQuote: true,
-        printWidth: 999,
-      },
+        trailingComma: "all",
+        printWidth: 140,
+        bracketSameLine: true,
+        plugins: ["@serverless-guru/prettier-plugin-import-order"],
```
Comment (BalbinaK) [inline index.js]: Do we need to install this plugin?

**Review by BalbinaK — COMMENTED** [review]
There's quite a few things added here that I'd need to dive deeper to understand why we need them, but I'm gonna focus on the one you mentioned in the PR description 😄 

>  plugin that will offer the possibility to add rules like "importOrderCaseInsensitive": true

What would be a use case for that so I better understand the reason why we need it? By looking at the [trivago/prettier-plugin-sort-imports docs](https://github.com/trivago/prettier-plugin-sort-imports?tab=readme-ov-file#importordercaseinsensitive) it seems to me like setting that rule to `true` would make upper-case files higher up in the order than lower-case ones:

Code Example with `importOrderCaseInsensitive: false`
```
import React from 'react';
import Button from './components/Button';
import apple from './fruits/Apple';
import Banana from './fruits/Banana';
```

Code Example with `importOrderCaseInsensitive: true`
```
import apple from './fruits/Apple';
import Banana from './fruits/Banana';
import Button from './components/Button';
import React from 'react';

**Comment by felicia-haggqvist** [comment]
@BalbinaK, I was talking to @imprashast and he recommended how another finn team's prettier-config looked like, where they were using `@serverless-guru/prettier-plugin-import-order` and said that it was a nice plugin. So I think it is just a preference on how we want to format it. But if we don't want to use the `@serverless-guru/prettier-plugin-import-order` and add the `importOrder`:s, we can remove them and discuss it later if we want to add them to our eslint-config :)

File: index.js (hunk 1 above)
Comment (felicia-haggqvist) [reply index.js]: @imprashast  Would we need to install this plugin in this @warp-ds/eslint-config-repo, or is it enough that the repos that use the `@warp-ds/eslint-config` install it as a devDep? When I linked this branch with the react-repo, I had to add it as a devDep and install it in the react-project.

Comment (imprashast) [reply index.js]: This can be added to `peerDependencies` if you'd like :)
Basically anything this repo needs as a dependency in another repo using it should be added to this repo's peerDeps :)

**Comment by BalbinaK** [comment]
> @BalbinaK, I was talking to @imprashast and he recommended how another finn team's prettier-config looked like, where they were using `@serverless-guru/prettier-plugin-import-order` and said that it was a nice plugin. 

A nice plugin you say 😄 That might be true, though I'm a bit hesitant to add yet another dependency to maintain to all Warp repos, seeing how difficult it is for us to maintain all of them on a regular basis ;) If however we want to add stuff like that, it would help if we understood what those prettier rules do. A link to that Finn project might help, I guess? :) 

> So I think it is just a preference on how we want to format it. But if we don't want to use the `@serverless-guru/prettier-plugin-import-order` and add the `importOrder`:s, we can remove them and discuss it later if we want to add them to our eslint-config :)

Maybe we want that, hard to say without understanding what the outcome of the above changes is 😄 This may be not worth a big discussion, but file formatting sounds like something a team should agree on. So maybe @magnuh and @Skadefryd would like to have a look at these changes, too, before they are implemented in all Warp repos 😊

**Comment by felicia-haggqvist** [comment]
> > @BalbinaK, I was talking to @imprashast and he recommended how another finn team's prettier-config looked like, where they were using `@serverless-guru/prettier-plugin-import-order` and said that it was a nice plugin.
> 
> A nice plugin you say 😄 That might be true, though I'm a bit hesitant to add yet another dependency to maintain to all Warp repos, seeing how difficult it is for us to maintain all of them on a regular basis ;) If however we want to add stuff like that, it would help if we understood what those prettier rules do. A link to that Finn project might help, I guess? :)
>
Here is the link to the Finn-project's prettier-config: https://github.schibsted.io/finn/realestate-item-page/blob/main/.prettierrc and you can read [here](https://github.com/serverless-guru/prettier-import-order) what prettier rules you can use when @serverless-guru/prettier-plugin-import-order is added 

> > So I think it is just a preference on how we want to format it. But if we don't want to use the `@serverless-guru/prettier-plugin-import-order` and add the `importOrder`:s, we can remove them and discuss it later if we want to add them to our eslint-config :)
> 
> Maybe we want that, hard to say without understanding what the outcome of the above changes is 😄 This may be not worth a big discussion, but file formatting sounds like something a team should agree on. So maybe @magnuh and @Skadefryd would like to have a look at these changes, too, before they are implemented in all Warp repos 😊
>
Sure, we can wait and see what Magnus and Snorre thinks :)

File: index.js (hunk 1 above)
Comment (magnuh) [reply index.js]: Does this solve anything that can't be solved with: https://eslint.org/docs/latest/rules/sort-imports ? 🙂

Comment (felicia-haggqvist) [reply index.js]: Hmm good question! 🤔 Need to check what differs between them!

Comment (felicia-haggqvist) [reply index.js]: If I’ve understood it correctly, the `@serverless-guru/prettier-plugin-import-order` integrates with the Prettier formatter to automatically reorder imports during code formatting with minimal configurability to help maintain simplicity and consistency in style. While EsLint’s `sort-imports` offers more detailed configurability and enforces import order, which offers more control over code standards and styles. So I guess it depends on what we prioritize. If we want to prioritize code formatting with minimal setup, then it sounds like `@serverless-guru/prettier-plugin-import-order` could work for us, or if we want to prioritize style and quality enforcement then we could go with EsLint's `sort-imports`. If we’re already using EsLint for linting then maybe it is easier to incorporate `sort-imports`, but it sounds like it is just a matter of preference. 🤔  @imprashast Correct me if I'm wrong here! 😄

**Comment by BalbinaK** [comment]
Thanks a lot for describing what the plugin controls with the suggested rules 🤗 

**tldr;** It looks like we can't avoid having yet another peerDependency to fix import order, so we can give this prettier plugin a try and see if it ever becomes a problem in the future. It would however be nice to have react/vue/lit imports in separate groups at the top in my opinion + maybe have local/parent dir imports separated, too.
I also looked at eslint-plugin-import to see if we could use that instead, since it's already a dep in @warp-ds/react, but though it fixes react and parent dir imports nicely, it does not fix types order and duplicate imports.

---

What I'm not sure about in this setup, now that I tested it in @warp-ds/react, is that imports from react are pushed down in the import order when external packages are present. Maybe there's a way for us to treat those separately if that makes sense to others? See Modal's `component.tsx` file:

```
import { classNames } from '@chbphone55/classnames';
import { i18n } from '@lingui/core';
import { modal as ccModal } from '@warp-ds/css/component-classes';
import IconArrowLeft16 from '@warp-ds/icons/react/arrow-left-16';
import IconClose16 from '@warp-ds/icons/react/close-16';
import React, { useEffect, useRef } from 'react';
import FocusLock from 'react-focus-lock/dist/es2015';
import { setup, teardown } from 'scroll-doctor';

import { activateI18n } from '../../i18n.js';
import { useId } from '../../utils/src/index.js';
import { messages as enMessages } from './locales/en/messages.mjs';
import { messages as fiMessages } from './locales/fi/messages.mjs';
import { messages as nbMessages } from './locales/nb/messages.mjs';

import type { ModalProps } from './props.js';
```

Also, it seems that any imports from parent or local directories are grouped together, which might be fine, but not sure is what we want.

On another note, I was wondering if there were other plugins that are more broadly used and could potentially be considered here, and the only one that made sense to me was `eslint-plugin-import` (which is already a dependency in @warp-ds/react as it looks like) with e.g. its [import/order](https://github.com/import-js/eslint-plugin-import/blob/c0ac54b8a721c2b1c9048838acc4d6282f4fe7a7/docs/rules/order.md) rule. There with the help of ChatGPT I tested the following config:

```
    "import/order": ["error", {
      "groups": ["builtin", "external", "parent", "sibling", "index"],
      "pathGroups": [
        {
          "pattern": "{react,react-dom/**}",
          "group": "external",
          "position": "before"
        }
      ],
      "pathGroupsExcludedImportTypes": ["react"],
      "newlines-between": "always",
      "alphabetize": {
        "order": "asc",
        "caseInsensitive": true
      }
    }],
    "import/newline-after-import": ["error", {
      "count": 1
    }],
    "import/first": "error",
    "import/no-duplicates": "error"
```
    
And the output in the same component file was as follows:

```
import React, { useEffect, useRef } from 'react';

import { classNames } from '@chbphone55/classnames';
import { i18n } from '@lingui/core';
import { modal as ccModal } from '@warp-ds/css/component-classes';
import IconArrowLeft16 from '@warp-ds/icons/react/arrow-left-16';
import IconClose16 from '@warp-ds/icons/react/close-16';
import FocusLock from 'react-focus-lock/dist/es2015';
import { setup, teardown } from 'scroll-doctor';

import { activateI18n } from '../../i18n.js';
import { useId } from '../../utils/src/index.js';

import { messages as enMessages } from './locales/en/messages.mjs';
import { messages as fiMessages } from './locales/fi/messages.mjs';
import { messages as nbMessages } from './locales/nb/messages.mjs';
import type { ModalProps } from './props.js';
```

**Limitations**
The things that this plugin would not do even with [@typescript-eslint/eslint-plugin](https://typescript-eslint.io/packages/eslint-plugin/), is push all type imports to the bottom and add a new-line before them.
Also, there's no real equivalent of `importOrderMergeDuplicateImports` - ESLint will warn about duplicate imports with `import/no-duplicates`, but it won't automatically merge them.

**Conclusion**
Now that I see that there's no way to avoid having yet another peerDependency to fix import order, I'm fine with trying out the suggested plugin 😊 We can always move away from it if it proves problematic with time.

**Comment by felicia-haggqvist** [comment]
>
> **tldr;** It looks like we can't avoid having yet another peerDependency to fix import order, so we can give this prettier plugin a try and see if it ever becomes a problem in the future. It would however be nice to have react/vue/lit imports in separate groups at the top in my opinion + maybe have local/parent dir imports separated, too. I also looked at eslint-plugin-import to see if we could use that instead, since it's already a dep in @warp-ds/react, but though it fixes react and parent dir imports nicely, it does not fix types order and duplicate imports.
>
I also looked into elsint-plugin-import, but found out what you just mentioned regarding types order and duplicate imports. 👍 So still sounds like maybe we should try out the @serverless-guru/prettier-import-order plugin. I can check out how/if we can change the rules for @serverless-guru/prettier-import-order plugin so that it separates and places react/vue/lit imports at the top and separate local/parent dir imports. :)

**Comment by felicia-haggqvist** [comment]
@BalbinaK I managed to tweak the `@serverless-guru/prettier-import-order` to format the imports like this:
```````
import React, { useEffect, useRef } from 'react';

import { classNames } from '@chbphone55/classnames';
import { i18n } from '@lingui/core';
import { modal as ccModal } from '@warp-ds/css/component-classes';
import IconArrowLeft16 from '@warp-ds/icons/react/arrow-left-16';
import IconClose16 from '@warp-ds/icons/react/close-16';
import FocusLock from 'react-focus-lock/dist/es2015';
import { setup, teardown } from 'scroll-doctor';

import { activateI18n } from '../../i18n.js';
import { useId } from '../../utils/src/index.js';
import { messages as enMessages } from './locales/en/messages.mjs';
import { messages as fiMessages } from './locales/fi/messages.mjs';
import { messages as nbMessages } from './locales/nb/messages.mjs';

import type { ModalProps } from './props.js';
```````

That's the closest I have gotten (so far) with getting it to look similar to how you managed to format it using eslint-plugin-import. :) The only thing I was not able to do was to create a new line between `import { useId } from '../../utils/src/index.js';` and `import { messages as enMessages } from './locales/en/messages.mjs';`.

Not sure if this is okay? :)

**Review by BalbinaK — APPROVED** [review]
Looks good to me! 

> The only thing I was not able to do was to create a new line between import { useId } from '../../utils/src/index.js'; and import { messages as enMessages } from './locales/en/messages.mjs';.
>
> Not sure if this is okay? :)


That's fine! Thanks for trying 😊 The plugin groups those parent directories, so it should be fine even without the newline 👍

**Comment by felicia-haggqvist** [comment]
@BalbinaK @imprashast After not finding a work-around with using the @serverless-guru/prettier-plugin-import-order when trying to format the @warp-ds/vue-project, I propose that we instead use the `eslint-plugin-import` and format it as you, Balbina, had suggested previously: 
> 
> ```
>     "import/order": ["error", {
>       "groups": ["builtin", "external", "parent", "sibling", "index"],
>       "pathGroups": [
>         {
>           "pattern": "{react,react-dom/**}",
>           "group": "external",
>           "position": "before"
>         }
>       ],
>       "pathGroupsExcludedImportTypes": ["react"],
>       "newlines-between": "always",
>       "alphabetize": {
>         "order": "asc",
>         "caseInsensitive": true
>       }
>     }],
>     "import/newline-after-import": ["error", {
>       "count": 1
>     }],
>     "import/first": "error",
>     "import/no-duplicates": "error"
> ```
> 

But where I have also added support for "vue" and "lit".
I have linked this branch with @warp-ds/react, @warp-ds/vue and @warp-ds/elements and it seems to format the files consistently. :) 

I know that you, Balbina, mentioned that the `eslint-plugin-import` does not take into account type-imports or duplicated imports, but I still think that it is a good import-plugin. :) 

You can find the changes in this commit: [db65934](https://github.com/warp-ds/eslint-config/pull/1/commits/db6593437c975997e735919a2f68c476c2799a42)

**Review by BalbinaK — APPROVED** [review]
LGMT 😄
